### ISSUE TITLE
Examples are broken in 0.12.x: "name" and "value" are required for setHeader()

### ISSUE THREAD

**Issue by darylrobbins** [splunk/splunk-sdk-javascript#38]
When attempting to run examples and access a Splunk server under NodeJS 0.12.x, server crashes with:

```
_http_outgoing.js:333
    throw new Error('"name" and "value" are required for setHeader().');
          ^
Error: "name" and "value" are required for setHeader().
    at ClientRequest.OutgoingMessage.setHeader (_http_outgoing.js:333:11)
    at new ClientRequest (_http_client.js:101:14)
    at Object.exports.request (http.js:49:10)
    at Object.exports.request (https.js:136:15)
    at Request.start (/root/projects/splunk_javascript_sdk/splunk-sdk/node_modules/request/index.js:594:30)
    at Request.end (/root/projects/splunk_javascript_sdk/splunk-sdk/node_modules/request/index.js:1186:28)
    at /root/projects/splunk_javascript_sdk/splunk-sdk/node_modules/request/index.js:436:12
    at process._tickCallback (node.js:355:11)
```

This is caused by https://github.com/request/request/issues/1522.

Ref: Splunk Support Case 234530

**Comment by shakeelmohamed** [comment]
@darylrobbins Thanks for bringing this to our attention

**Comment by darylrobbins** [comment]
Sorry, but I am going to have to leave merging the fix with the development branch in your hands. As confirmed by support, the examples are broken under Node 0.12.x. (Please see my case 234530).

**Comment by shakeelmohamed** [comment]
Fixed! We just released a fix for this as part of v1.7.1
